freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/VirtualGroupSyncService.java

```diff
@@ -0,0 +1,52 @@
+package com.sequenceiq.freeipa.service.freeipa.user;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.auth.altus.UmsVirtualGroupRight;
+import com.sequenceiq.cloudbreak.auth.altus.VirtualGroupService;
+import com.sequenceiq.freeipa.configuration.UsersyncConfig;
+import com.sequenceiq.freeipa.service.freeipa.user.model.UserSyncOptions;
+
+@Service
+public class VirtualGroupSyncService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(VirtualGroupSyncService.class);
+
+    @Inject
+    @Qualifier(UsersyncConfig.USERSYNC_TASK_EXECUTOR)
+    private ExecutorService asyncTaskExecutor;
+
+    @Inject
+    private VirtualGroupService virtualGroupService;
+
+    public void syncVirtualGroups(String accountId, Set<String> environmentCrns, UserSyncOptions options) {
+        if (options.isFullSync()) {
+            LOGGER.info("Sync virtual groups for environments: {}", environmentCrns);
+            Map<String, Future<Map<UmsVirtualGroupRight, String>>> userGroupFutures = environmentCrns.stream().collect(Collectors.toMap(envCrn -> envCrn, envCrn -> asyncCreateUserGroups(accountId, envCrn)));
+            userGroupFutures.forEach((envCrn, userGroupFuture) -> {
+                try {
+                    Map<UmsVirtualGroupRight, String> virtualGroups = userGroupFuture.get();
+                    LOGGER.info("Virtual group sync finished for env '{}'. Synchronized virtual groups: '{}'", envCrn, virtualGroups.values());
+                } catch (Exception e) {
+                    LOGGER.error("Virtual group sync failed for env '{}': '{}'", envCrn, e.getMessage());
```

Comment:
instead of this use:
`LOGGER.error("Virtual group sync failed for env '{}''", envCrn);`
so we would have the message and the stacktrace in the logs

Reply:
replaced